### ISSUE TITLE
Cannot detect `missing-underscore-after-hyperlink` with line break

### ISSUE THREAD

**Issue by koyuki7w** [sphinx-contrib/sphinx-lint#140]
sphinx-lint does not detect `missing-underscore-after-hyperlink` when the link contains a newline:
```
`Link text
<https://example.com/>`
```

This seems to be because the `check_missing_underscore_after_hyperlink` function checks each line; it should (like other checker functions) check each paragraph.
